front/src/applications/operationalStudies/views/Scenario/components/MacroEditor/utils.ts

```
@@ -151,6 +151,99 @@ export const deleteMacroNodeByNgeId = async (
+      osrdEditoastApi.endpoints.postProjectsByProjectIdStudiesAndStudyIdScenariosScenarioIdMacroNotes.initiate(
+        {
+          projectId: state.projectId,
+          studyId: state.studyId,
+          scenarioId: state.scenarioId,
+          macroNoteBatchForm: {
+            macro_notes: [
+              {
+                x: Math.round(note.x),
+                y: Math.round(note.y),
+                title: note.title,
+                text: note.text,
+                labels: note.labelIds.map((id) => id.toString()),
+              },
+            ],
+          },
+        }
+      )
+    ).unwrap();
+
+    const createdNote = response.macro_notes[0];
+    state.updateNote(note.ngeId, { dbId: createdNote.id });
+  } catch (e) {
+    console.error(e);
+  }
+};
+
+export const updateMacroNote = async (
+  state: MacroEditorState,
+  dispatch: AppDispatch,
+  note: NoteIndexed
+) => {
+  try {
+    const indexedNote = state.getNoteByNgeId(note.ngeId);
+    if (!indexedNote) throw new Error(`Note ${note.ngeId} not found`);
+    if (!indexedNote.dbId) throw new Error(`Note ${note.ngeId} is not saved in the DB`);
+
+    await dispatch(
+      osrdEditoastApi.endpoints.putProjectsByProjectIdStudiesAndStudyIdScenariosScenarioIdMacroNotesNoteId.initiate(
+        {
+          projectId: state.projectId,
+          studyId: state.studyId,
+          scenarioId: state.scenarioId,
+          noteId: indexedNote.dbId,
+          macroNoteForm: {
+            x: Math.round(note.x),
+            y: Math.round(note.y),
+            title: note.title,
+            text: note.text,
+            labels: note.labelIds.map((id) => id.toString()),
+          },
+        }
+      )
+    ).unwrap();
+
+    state.updateNote(note.ngeId, note);
+  } catch (e) {
+    console.error(e);
+  }
+};
+
+export const deleteMacroNote = async (
+  state: MacroEditorState,
+  dispatch: AppDispatch,
+  note: NoteIndexed
+) => {
+  try {
+    if (note.dbId) {
```

We should throw an error if we're missing a DB ID, instead of silently doing nothing. This makes bugs easier to debug.

---

Done